diff --git a/lib/common/noise/Noise.cc b/lib/common/noise/Noise.cc
index b64bae3..9901498 100644
--- a/lib/common/noise/Noise.cc
+++ b/lib/common/noise/Noise.cc
@@ -13,7 +13,7 @@ using namespace scene_rdl2::math;
 
 bool Noise::sNoiseIsDataInitialized = false;
 scene_rdl2::util::Random Noise::sNoiseRandom = scene_rdl2::util::Random(0xbeadceef);
-tbb::mutex Noise::sNoiseInitDataMutex;
+std::mutex Noise::sNoiseInitDataMutex;
 std::vector<int> Noise::sNoisePermutationTable;
 
 Noise::Noise(const int seed,
@@ -25,7 +25,7 @@ Noise::Noise(const int seed,
     mIspc.mTableSize = tableSize;
 
     if (useStaticTables) {
-        tbb::mutex::scoped_lock dataLock(sNoiseInitDataMutex); // Lock this part that initializes static data
+        std::scoped_lock dataLock(sNoiseInitDataMutex); // Lock this part that initializes static data
         // Only construct the static data once to share across all instances
         if (!sNoiseIsDataInitialized) {
             buildStaticTables(tableSize);
diff --git a/lib/common/noise/Noise.h b/lib/common/noise/Noise.h
index 27e8670..826b9a2 100644
--- a/lib/common/noise/Noise.h
+++ b/lib/common/noise/Noise.h
@@ -8,10 +8,11 @@
 #pragma warning disable 1711 // Warnings about assignemnt to statically allocated data
 
 #include "Noise_ispc_stubs.h"
-#include <tbb/mutex.h>
 #include <scene_rdl2/common/math/Vec3.h>
 #include <scene_rdl2/common/math/Xform.h>
 
+#include <mutex>
+
 // Forward declaration
 namespace scene_rdl2 {
 namespace util {
@@ -71,7 +72,7 @@ protected:
     // Static data
     static bool sNoiseIsDataInitialized;
     static scene_rdl2::util::Random sNoiseRandom;
-    static tbb::mutex sNoiseInitDataMutex;
+    static std::mutex sNoiseInitDataMutex;
     static std::vector<int> sNoisePermutationTable;
 
 private:
diff --git a/lib/common/noise/Worley.cc b/lib/common/noise/Worley.cc
index 1fd33e8..522240c 100644
--- a/lib/common/noise/Worley.cc
+++ b/lib/common/noise/Worley.cc
@@ -227,7 +227,7 @@ collectPoints(const Worley_PointArray::iterator tmpWorleyPointsBeg,
 
 // Static data
 bool Worley::sIsWorleyDataInitialized = false;
-tbb::mutex Worley::sWorleyInitDataMutex;
+std::mutex Worley::sWorleyInitDataMutex;
 std::vector<float> Worley::sWorleyPointsX;
 std::vector<float> Worley::sWorleyPointsY;
 std::vector<float> Worley::sWorleyPointsZ;
@@ -255,7 +255,7 @@ Worley::Worley(const int seed,
     initPointProbabilities();
 
     if (useStaticTables) {
-        tbb::mutex::scoped_lock dataLock(sWorleyInitDataMutex); // Lock this part that initializes static data
+        std::scoped_lock dataLock(sWorleyInitDataMutex); // Lock this part that initializes static data
         // Only construct the static data once to share across all instances
         if (!sIsWorleyDataInitialized) {
             buildStaticPointTables(addNormals, tableSize);
diff --git a/lib/common/noise/Worley.h b/lib/common/noise/Worley.h
index 741450a..0a29231 100644
--- a/lib/common/noise/Worley.h
+++ b/lib/common/noise/Worley.h
@@ -7,6 +7,8 @@
 
 #pragma warning disable 1711 // Warnings about assignemnt to statically allocated data
 
+#include <array>
+
 #include "Noise.h"
 #include "Noise_ispc_stubs.h"
 #include "Worley_ispc_stubs.h"
@@ -184,7 +186,7 @@ private:
 
     // Static data
     static bool sIsWorleyDataInitialized;
-    static tbb::mutex sWorleyInitDataMutex;
+    static std::mutex sWorleyInitDataMutex;
 
     static std::vector<float> sWorleyPointsX;
     static std::vector<float> sWorleyPointsY;
