diff --git a/cmd/moonray_gui/RenderGui.h b/cmd/moonray_gui/RenderGui.h
index 526ac1b..e6e707a 100644
--- a/cmd/moonray_gui/RenderGui.h
+++ b/cmd/moonray_gui/RenderGui.h
@@ -120,7 +120,7 @@ private:
     /// The renderering code will strive to render this frame. If it's rendering
     /// a frame with a lower timestamp then we know the frame it's currently
     /// rendering is old.
-    tbb::atomic<uint32_t>   mMasterTimestamp;
+    std::atomic<uint32_t>   mMasterTimestamp;
 
     /// The timestamp of the frame the renderer is currently processing.
     uint32_t                mRenderTimestamp;
