diff --git a/openvdb/openvdb/tools/PointIndexGrid.h b/openvdb/openvdb/tools/PointIndexGrid.h
index c18f3668..bae5c025 100644
--- a/openvdb/openvdb/tools/PointIndexGrid.h
+++ b/openvdb/openvdb/tools/PointIndexGrid.h
@@ -1427,7 +1427,7 @@ public:
     bool operator!=(const PointIndexLeafNode& other) const { return !(other == *this); }
 
     template<MergePolicy Policy> void merge(const PointIndexLeafNode& rhs) {
-        BaseLeaf::merge<Policy>(rhs);
+        BaseLeaf::template merge<Policy>(rhs);
     }
     template<MergePolicy Policy> void merge(const ValueType& tileValue, bool tileActive) {
          BaseLeaf::template merge<Policy>(tileValue, tileActive);
