diff --git a/cmake/defaults/Packages.cmake b/cmake/defaults/Packages.cmake
index 825c5af1f..fe1510cdb 100644
--- a/cmake/defaults/Packages.cmake
+++ b/cmake/defaults/Packages.cmake
@@ -37,71 +37,71 @@ set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 find_package(Threads REQUIRED)
 set(PXR_THREAD_LIBS "${CMAKE_THREAD_LIBS_INIT}")
 
-# Find Boost package before getting any boost specific components as we need to
-# disable boost-provided cmake config, based on the boost version found.
-find_package(Boost REQUIRED)
-
-# Boost provided cmake files (introduced in boost version 1.70) result in 
-# inconsistent build failures on different platforms, when trying to find boost 
-# component dependencies like python, program options, etc. Refer some related 
-# discussions:
-# https://github.com/boostorg/python/issues/262#issuecomment-483069294
-# https://github.com/boostorg/boost_install/issues/12#issuecomment-508683006
-#
-# Hence to avoid issues with Boost provided cmake config, Boost_NO_BOOST_CMAKE
-# is enabled by default for boost version 1.70 and above. If a user explicitly 
-# set Boost_NO_BOOST_CMAKE to Off, following will be a no-op.
-if (${Boost_VERSION_STRING} VERSION_GREATER_EQUAL "1.70")
-    option(Boost_NO_BOOST_CMAKE "Disable boost-provided cmake config" ON)
+if(PXR_ENABLE_OPENVDB_SUPPORT)
+    # Find Boost package before getting any boost specific components as we need to
+    # disable boost-provided cmake config, based on the boost version found.
+    find_package(Boost REQUIRED)
+    # If a user explicitly sets Boost_NO_BOOST_CMAKE to On, following will
+    # disable the use of boost provided cmake config.
+    option(Boost_NO_BOOST_CMAKE "Disable boost-provided cmake config" OFF)
     if (Boost_NO_BOOST_CMAKE)
-        message(STATUS "Disabling boost-provided cmake config")
+      message(STATUS "Disabling boost-provided cmake config")
     endif()
 endif()
 
 if(PXR_ENABLE_PYTHON_SUPPORT)
     # --Python.
-    if(PXR_USE_PYTHON_3)
-        find_package(PythonInterp 3.0 REQUIRED)
-        find_package(PythonLibs 3.0 REQUIRED)
-    else()
-        find_package(PythonInterp 2.7 REQUIRED)
-        find_package(PythonLibs 2.7 REQUIRED)
-    endif()
+    macro(setup_python_package package)
+        find_package(${package} COMPONENTS Interpreter Development REQUIRED)
+
+        # Set up versionless variables so that downstream libraries don't
+        # have to worry about which Python version is being used.
+        set(PYTHON_EXECUTABLE "${${package}_EXECUTABLE}")
+        set(PYTHON_INCLUDE_DIRS "${${package}_INCLUDE_DIRS}")
+        set(PYTHON_VERSION_MAJOR "${${package}_VERSION_MAJOR}")
+        set(PYTHON_VERSION_MINOR "${${package}_VERSION_MINOR}")
+
+        # Convert paths to CMake path format on Windows to avoid string parsing
+        # issues when we pass PYTHON_EXECUTABLE or PYTHON_INCLUDE_DIRS to
+        # pxr_library or other functions.
+        if(WIN32)
+            file(TO_CMAKE_PATH ${PYTHON_EXECUTABLE} PYTHON_EXECUTABLE)
+            file(TO_CMAKE_PATH ${PYTHON_INCLUDE_DIRS} PYTHON_INCLUDE_DIRS)
+        endif()
 
-    if(WIN32 AND PXR_USE_DEBUG_PYTHON)
-        set(Boost_USE_DEBUG_PYTHON ON)
-    endif()
+        # PXR_PY_UNDEFINED_DYNAMIC_LOOKUP might be explicitly set when 
+        # packaging wheels, or when cross compiling to a Python environment 
+        # that is not the current interpreter environment.
+        # If it was not explicitly set to ON or OFF, then determine whether 
+        # Python was statically linked to its runtime library by fetching the
+        # sysconfig variable LDLIBRARY, and set the variable accordingly.
+        # If the variable does not exist, PXR_PY_UNDEFINED_DYNAMIC_LOOKUP will
+        # default to OFF. On Windows, LDLIBRARY does not exist, as the default
+        # will always be OFF.
+        if((NOT WIN32) AND (NOT DEFINED PXR_PY_UNDEFINED_DYNAMIC_LOOKUP))
+            execute_process(COMMAND ${PYTHON_EXECUTABLE} "-c" "import sysconfig;print(sysconfig.get_config_var('LDLIBRARY'))"
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+                OUTPUT_VARIABLE PXR_PYTHON_LINKED_LIBRARY
+            )
+            get_filename_component(PXR_PYTHON_LINKED_LIBRARY_EXT ${PXR_PYTHON_LINKED_LIBRARY} LAST_EXT)
+            if(PXR_PYTHON_LINKED_LIBRARY_EXT STREQUAL ".a")
+                set(PXR_PY_UNDEFINED_DYNAMIC_LOOKUP ON)
+                message(STATUS 
+                        "PXR_PY_UNDEFINED_DYNAMIC_LOOKUP wasn't specified, forced ON because Python statically links ${PXR_PYTHON_LINKED_LIBRARY}")
+            endif()
+        endif()
 
-    # This option indicates that we don't want to explicitly link to the python
-    # libraries. See BUILDING.md for details.
-    if(PXR_PY_UNDEFINED_DYNAMIC_LOOKUP AND NOT WIN32 )
-        set(PYTHON_LIBRARIES "")
-    endif()
+        # This option indicates that we don't want to explicitly link to the
+        # python libraries. See BUILDING.md for details.
+        if(PXR_PY_UNDEFINED_DYNAMIC_LOOKUP AND NOT WIN32)
+            set(PYTHON_LIBRARIES "")
+        else()
+            set(PYTHON_LIBRARIES "${package}::Python")
+        endif()
+    endmacro()
 
-    if (${Boost_VERSION_STRING} VERSION_GREATER_EQUAL "1.67")
-        # As of boost 1.67 the boost_python component name includes the
-        # associated Python version (e.g. python27, python36). 
-        # XXX: After boost 1.73, boost provided config files should be able to 
-        # work without specifying a python version!
-        # https://github.com/boostorg/boost_install/blob/master/BoostConfig.cmake
-
-        # Find the component under the versioned name and then set the generic
-        # Boost_PYTHON_LIBRARY variable so that we don't have to duplicate this
-        # logic in each library's CMakeLists.txt.
-        set(python_version_nodot "${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
-        find_package(Boost
-            COMPONENTS
-                python${python_version_nodot}
-            REQUIRED
-        )
-        set(Boost_PYTHON_LIBRARY "${Boost_PYTHON${python_version_nodot}_LIBRARY}")
-    else()
-        find_package(Boost
-            COMPONENTS
-                python
-            REQUIRED
-        )
-    endif()
+    # USD builds only work with Python3
+    setup_python_package(Python3)
 
     # --Jinja2
     find_package(Jinja2)
diff --git a/cmake/modules/FindTBB.cmake b/cmake/modules/FindTBB.cmake
index 9bf69a022..73a9ceafb 100644
--- a/cmake/modules/FindTBB.cmake
+++ b/cmake/modules/FindTBB.cmake
@@ -197,7 +197,12 @@ if(NOT TBB_FOUND)
   ##################################
 
   if(TBB_INCLUDE_DIRS)
-    file(READ "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h" _tbb_version_file)
+    # Use new oneTBB version header if it exists.
+    if(EXISTS "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h")
+      file(READ "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h" _tbb_version_file)
+    else()
+      file(READ "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h" _tbb_version_file)
+    endif()
     string(REGEX REPLACE ".*#define TBB_VERSION_MAJOR ([0-9]+).*" "\\1"
         TBB_VERSION_MAJOR "${_tbb_version_file}")
     string(REGEX REPLACE ".*#define TBB_VERSION_MINOR ([0-9]+).*" "\\1"
diff --git a/pxr/base/gf/plane.h b/pxr/base/gf/plane.h
index 61c731085..f4df8796e 100644
--- a/pxr/base/gf/plane.h
+++ b/pxr/base/gf/plane.h
@@ -32,6 +32,7 @@
 #include "pxr/base/gf/api.h"
 
 #include <iosfwd>
+#include <vector>
 
 PXR_NAMESPACE_OPEN_SCOPE

diff --git a/pxr/usd/sdf/path.h b/pxr/usd/sdf/path.h
index a0986d26d..e4577e000 100644
--- a/pxr/usd/sdf/path.h
+++ b/pxr/usd/sdf/path.h
@@ -306,9 +306,6 @@ public:
     /// Constructs the default, empty path.
     ///
     SdfPath() noexcept {
-        // This generates a single instruction instead of 2 on gcc 6.3.  Seems
-        // to be fixed on gcc 7+ and newer clangs.  Remove when we're there!
-        memset(this, 0, sizeof(*this));
     }
 
     /// Creates a path from the given string.
